objects/src/testing/storage.rs

```diff
@@ -33,21 +33,24 @@ impl AccountStorageBuilder {
     pub fn new() -> Self {
         Self {
             slots: vec![
-                AccountStorage::mock_item_2().slot,
                 AccountStorage::mock_item_0().slot,
                 AccountStorage::mock_item_1().slot,
+                AccountStorage::mock_item_2().slot,
             ],
         }
     }
 
-    pub fn add_slot(&mut self, slot: StorageSlot) -> &mut Self {
-        self.slots.push(slot);
+    pub fn add_slot(&mut self, slot: StorageSlot, location: Option<usize>) -> &mut Self {
```

Comment:
Could we document what `location` does here because I think the behavior may not be intuitive.

But more generally, why do we need to insert slots in front of other slots?

Reply:
# Explanation of Public Key Storage in Tests

## Background
Some tests generate a `pub_key` that needs to be added to account storage. 

## Previous Approach

In the past, we could add the `pub_key` to a specific location in storage:
- We used the leaf index in the storage Merkle tree to place it at position 0.

## Current Challenge
We can no longer rearrange storage to put the generated key at location 0. The default behaviour is pushing to the storage array.

## Implications
- Without a `location` argument, we would default to pushing the `pub_key` to location x.
- This would break tests expecting the `pub_key` to be at location 0.

## Solution
We introduce an additional `location` argument to:
- Allow placing the generated `pub_key` at the needed location.
- Maintain compatibility with existing tests.

Reply:
@bobbinth let me know what you think.

Reply:
For the tests you mention, what is already in slot 0 when we try to add `pub_key` to storage? (i.e, whey do we have slot 0 already initialized with something before adding `pub_key` to it).

Reply:
**We use `location` in 4 places**:

1. We use it here to modify the standard behaviour and place the `faucet metadata` at the right storage location: 

https://github.com/0xPolygonMiden/miden-base/pull/875/files#diff-4a16cc24ae5bdf839333a6138a854d0245bff2c3f569047d0ba74bf17377f274R487-R497

2. Same here:

https://github.com/0xPolygonMiden/miden-base/pull/875/files#diff-79790044ea5a3638d7b1ba506a652d060c3ba57517b8989f08a9a7b70f85b999R337-R363

3. Here we use it in `build_with_auth` to place the `pub_key` at the right location in storage as you can see here a `StorageBuilder` is provided that is already instantiated with storage slots: 

https://github.com/0xPolygonMiden/miden-base/pull/875/files#diff-d1f8b21797d4f659cb178e5db9a0ff0268125659de672ce3e1415a6f650ee91bR143-R159

4. Lastly, we use it in `create_new_account` helper function once again to add the `pub_key` at location 0:

https://github.com/0xPolygonMiden/miden-base/pull/875/files#diff-8c72099bb8905470b853de8e3fffe9717e70a5a717c34bd6ecf664bd44588df6R355-R369

Looking into the 4 uses of `location` it seems like we could refactor to not have to add it. But this would come with some refactor of how we mock storage.

Reply:
Ah - I see: `AccountStorageBuilder` always populates the first 3 storage slots with some dummy values, right?

Maybe the right approach here is to do the following:

- Rename the current `AccountStorageBuilder::new()` into `AccountStorageBuilder::with_mock_data()`.
- Add `AccountStorageBuilder::new()` which creates an empty `AccountStorageBuilder`.
- Add `AccountBuilder::with_mock_storage()` which would work like the current `AccountBuilder::new()`(i.e., it would internally use `AccountStorageBuilder::with_mock_data()`).

Reply:
I'm not sure when that was changed but I was changing it back to be clear on a `AccountStorageBuilder::new()` on my PR as well (#871)